LPFilter/Source/PluginProcessor.cpp

```diff
@@ -167,15 +176,36 @@ void LpfilterAudioProcessor::processBlock (AudioSampleBuffer& ioBuffer, MidiBuff
     }
 }
 
-void LpfilterAudioProcessor::process(dsp::ProcessContextReplacing<float> context) noexcept
+void LpfilterAudioProcessor::process (AudioSampleBuffer& processBuffer) noexcept
```

Comment:
This process method is a bit complicated. You're trying to do mono and stereo processing and call `lpfDspLib->process` (which has 2 channels). 
You could write the following

```c++
    int bufferChans = processBuffer.getNumChannels(); // channels of the input buffer (1 or 2)
    int filterChans = tempBuffer.getNumChannels(); // channels of the filtered buffer (2)
    
    for (int iChan = 0; iChan < filterChans; ++iChan)
    {
         /* left channel of tempBuffer = left channel of processBuffer
         right channel of tempBuffer = left channel of processBuffer(mono version) 
                                                or right channel of processBuffer(stereo version) 
         */
        tempBuffer.copyFrom(iChan, 0, processBuffer, jmin(iChan, bufferChans - 1), 0, processBuffer.getNumSamples());
    }

    // process data
    lpfDspLib->process(tempBuffer.getNumSamples(), tempBuffer.getArrayOfWritePointers());
    
    for (int iChan = 0; iChan < bufferChans; ++iChan)
    {
        // copy data to the output buffer
        processBuffer.copyFrom(iChan, 0, tempBuffer, iChan, 0, processBuffer.getNumSamples());
    }
```